Cabal/Distribution/Simple/Utils.hs

```diff
@@ -1415,14 +1415,14 @@ dropExeExtension filepath =
     (filepath', extension) | extension `elem` exeExtensions -> filepath'
                            | otherwise                      -> filepath
 
--- | List of possible executable file extensions on the current platform.
+-- | List of possible executable file extensions on the current build platform.
 exeExtensions :: [String]
 exeExtensions = case buildOS of
   -- Possible improvement: on Windows, read the list of extensions from the
   -- PATHEXT environment variable. By default PATHEXT is ".com; .exe; .bat;
   -- .cmd".
-  Windows -> ["", "exe"]
-  Ghcjs   -> ["", "exe"]
+  Windows -> ["", ".exe"]
+  Ghcjs   -> ["", ".exe"]
```

Comment:
So technically this is breaking API compatibility since this is exported, but there don't seem to be any usages outside of Cabal oh hackage: https://codesearch.aelve.com/haskell/search?query=exeExtensions&insensitive=off&space=off&precise=on&sources=on

Furthermore `<.>` doesn't care about the leading dot in the extension, so `"foo" <.> "bar" == "foo" <.> ".bar" == "foo.bar"`.